src/Pim/Bundle/CatalogBundle/tests/integration/ElasticSearch/IndexConfiguration/PimCatalogNumberIntegration.php

```diff
@@ -0,0 +1,302 @@
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_1', 'product_2', 'product_4', 'product_5', 'product_6']
+        );
+    }
+
+    public function testGreaterThanOrEqualsOperator()
+    {
+        $query = [
+            'query' => [
+                'bool' => [
+                    'filter' => [
+                        'range' => [
+                            'values.box_quantity-decimal.<all_locales>.<all_channels>' => ['gte' => 10],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts($productsFound, ['product_1', 'product_3', 'product_4']);
+    }
+
+    public function testGreaterThanOperator()
+    {
+        $query = [
+            'query' => [
+                'bool' => [
+                    'filter' => [
+                        'range' => [
+                            'values.box_quantity-decimal.<all_locales>.<all_channels>' => ['gt' => 10],
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts($productsFound, ['product_3', 'product_4']);
+    }
+
+    public function testEmptyOperator()
+    {
+        $query = [
+            'query' => [
+                'bool' => [
+                    'must_not' => [
+                        'exists' => [
+                            'field' => 'values.box_quantity-decimal.<all_locales>.<all_channels>',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts($productsFound, ['product_7']);
+    }
+
+    public function testNotEmptyOperator()
+    {
+        $query = [
+            'query' => [
+                'bool' => [
+                    'filter' => [
+                        'exists' => [
+                            'field' => 'values.box_quantity-decimal.<all_locales>.<all_channels>',
+                        ],
+                    ],
+                ],
+            ],
+        ];
+
+        $productsFound = $this->getSearchQueryResults($query);
+
+        $this->assertProducts(
+            $productsFound,
+            ['product_1', 'product_2', 'product_3', 'product_4', 'product_5', 'product_6']
+        );
+    }
+
+    public function testSortAscending()
+    {
+        $query = [
+            'query' => [
+                'match_all' => new \stdClass(),
+            ],
+            'sort'  => [
+                [
+                    'values.box_quantity-decimal.<all_locales>.<all_channels>' => [
+                        'order'   => 'asc',
+                        'missing' => '_first',
```

Comment:
why should the products which are missing the field "box_quantity" should appeared first in the results?